regression/apps/getpid/getpid.c

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <sys/syscall.h>
+#include <time.h>
+#include <unistd.h>
+
+#define NUM_OF_CALLS 10000000
+
+int main()
+{
+	struct timespec start, end;
+	double seconds, nanoseconds;
+	double total_time, syscalls_per_second;
+	pid_t pid;
+
+	clock_gettime(CLOCK_MONOTONIC, &start);
+
+	for (int i = 0; i < NUM_OF_CALLS; i++) {
+		pid = syscall(SYS_getpid);
+	}
+
+	clock_gettime(CLOCK_MONOTONIC, &end);
+
+	seconds = (float)(end.tv_sec - start.tv_sec);
+	nanoseconds = (float)(end.tv_nsec - start.tv_nsec);
+	if (nanoseconds < 0) {
+		seconds--;
+		nanoseconds += 1e9;
+	}
+
+	total_time = seconds + nanoseconds / 1e9;
+	syscalls_per_second = NUM_OF_CALLS / total_time;
+
+	printf("Process %d executed the getpid() syscall %d times.\n", pid,
+	       NUM_OF_CALLS);
+	printf("Elapsed time: %.9lf seconds\n", total_time);
+	printf("Syscalls per second: %0.lf\n", syscalls_per_second);
```

Comment:
I think measuring `getpid`'s the latency (in nanoseconds) gives us a better sense of how much _time_ is spent on the system call than the throughput.

Reply:
Done